File: app/src/main/java/hottopic/mit/co/nz/cleaningservice/network/ServiceTypesRequest.java
package hottopic.mit.co.nz.cleaningservice.network;

import android.content.Context;

import java.util.Map;

import hottopic.mit.co.nz.cleaningservice.entities.network.response.ServiceTypesResponse;
import hottopic.mit.co.nz.cleaningservice.network.service.OrderService;
import io.reactivex.Single;

public class ServiceTypesRequest extends BaseRequest<OrderService, ServiceTypesResponse, ServiceTypesResponse> {
    private Context context;

    public ServiceTypesRequest(Context context) {
        super(context);
        this.context = context;
    }

    @Override
    protected Map<String, String> getHeader() {
        return null;
    }

    @Override
    protected Class<OrderService> getEndpointClass() {
        return OrderService.class;
    }

    @Override
    protected Single<ServiceTypesResponse> getEndpoint(OrderService endpoint) {
        return endpoint.getServiceTypes();
    }

    @Override
    protected ServiceTypesResponse dealWithResponse(ServiceTypesResponse response) {
        return response;
    }
}

File: app/src/main/java/hottopic/mit/co/nz/cleaningservice/network/ServiceTypesRequest.kt
package hottopic.mit.co.nz.cleaningservice.network

import android.content.Context

import hottopic.mit.co.nz.cleaningservice.entities.network.response.ServiceTypesResponse
import hottopic.mit.co.nz.cleaningservice.network.service.OrderService
import io.reactivex.Single

class ServiceTypesRequest(context: Context) : BaseRequest<OrderService, ServiceTypesResponse, ServiceTypesResponse>(context) {

    override val header: Map<String, String>?
        get() = null

    override val endpointClass: Class<OrderService>
        get() = OrderService::class.java

    override fun getEndpoint(endpoint: OrderService?): Single<ServiceTypesResponse?>? {
        return endpoint?.serviceTypes
    }

    override fun dealWithResponse(response: ServiceTypesResponse): ServiceTypesResponse? {
        return response
    }
}
